### PR TITLE
Add support for id property type

### PR DESCRIPTION
**Pull request by thetutlage**
FIXES CORE-2350

The `id` property is a special property that leads to generation of Mikro ORM hooks that runs on `onCreate` and `onInit`. 

The idea is to simplify the generation id and mark it is primary using a branded type.

### CODE REVIEW

File: packages/core/utils/package.json [hunk 1]
```diff
@@ -46,7 +46,7 @@
     "prepublishOnly": "cross-env NODE_ENV=production tsc --build",
     "build": "rimraf dist && tsc --build",
     "watch": "tsc --build --watch",
-    "test": "jest --silent --bail --maxWorkers=50% --forceExit --testPathIgnorePatterns='/integration-tests/' -- src/**/__tests__/**/*.ts",
+    "test": "jest --silent=false --bail --maxWorkers=50% --forceExit --testPathIgnorePatterns='/integration-tests/' -- src/**/__tests__/**/*.ts",
```
Comment (adrien2p) [inline packages/core/utils/package.json]: rm 😄

File: packages/core/utils/src/dml/helpers/create-mikro-orm-entity.ts [hunk 2]
```diff
@@ -170,6 +174,39 @@ export function createMikrORMEntity() {
       return
     }
 
+    /**
+     * Defining an id property
+     */
+    if (field.dataType.name === "id") {
+      const IdDecorator = field.dataType.options?.primaryKey
```
Comment (adrien2p) [inline packages/core/utils/src/dml/helpers/create-mikro-orm-entity.ts:181]: q: would it have the same effect to use the property decorator with primary: true?

File: packages/core/utils/src/dml/helpers/create-mikro-orm-entity.ts [hunk 3]
```diff
@@ -170,6 +174,39 @@ export function createMikrORMEntity() {
...
+            nullable: field.nullable,
+          })
+
+      IdDecorator(MikroORMEntity.prototype, field.fieldName)
+
+      /**
+       * Hook to generate entity within the code
+       */
+      MikroORMEntity.prototype.generateId = function () {
+        this.id = generateEntityId(this.id, "acc")
```
Comment (thetutlage) [inline packages/core/utils/src/dml/helpers/create-mikro-orm-entity.ts]: @adrien2p What should be the prefix of id generation. Do we follow some convention?

Comment (adrien2p) [inline packages/core/utils/src/dml/helpers/create-mikro-orm-entity.ts]: todo: the prefix should be configuration, also, if not provided maybe we can infer a default value that depends on the model name as well wdyt?

Comment (adrien2p) [reply packages/core/utils/src/dml/helpers/create-mikro-orm-entity.ts]: ahah I ve put a comment about it, we generally use a short name that correspond to the model name and that is meaningful a bit. so it should be configurable and if not provided we can infer it from the model name I would say. Something along the line: transform to snake case, split on `_`, if length 1 then the first 3 letters, if length 2 then the first two letters of each and if more then the first letter of each. wdyt?

File: packages/core/utils/package.json (hunk 1 above)
Comment (thetutlage) [reply packages/core/utils/package.json]: I think I should write a commit hook to fail if this substring exists

File: packages/core/utils/src/dml/helpers/create-mikro-orm-entity.ts (hunk 2 above)
Comment (thetutlage) [reply packages/core/utils/src/dml/helpers/create-mikro-orm-entity.ts:181]: No idea. I decided to stick with what rest of the codebase is using.

Comment (adrien2p) [reply packages/core/utils/src/dml/helpers/create-mikro-orm-entity.ts:181]: Lets go with that and de can revisit this later it is nit 😅

**Review by adrien2p — APPROVED** [review]

**Pull request merged**